react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {image: 0};
+    this.props = props;
+    this.handleClickLeft = this.handleClickLeft.bind(this);
+    this.handleClickRight = this.handleClickRight.bind(this);
+    this.handleDotClick = this.handleDotClick.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+  }
+
+  componentDidMount() {
+    this.int = setInterval(() => this.handleClickRight(), 3000)
+  }
+
+  render(props) {
+    let prop = this.props.poke;
+
+    return (
+      <div className="container">
+        <div className="carousel">
+          <div className="left">
+            <i className="fas fa-chevron-left" onClick={this.handleClickLeft}></i>
+          </div>
+          <div className="images">
+            <img src={prop.[this.state.image].link} />
+          </div>
+            <div className="right">
+              <i className="fas fa-chevron-right " onClick={this.handleClickRight}></i>
+            </div>
+          </div>
+          <div className="dots">
+            {
+            prop.map(item => {
+              return (
+                <i key={item.id} className={item.id === this.state.image ? 'fas fa-circle' : 'far fa-circle'} onClick={() => this.handleDotClick(item.id)}></i>
+              )
+              })
+            }
+          </div>
+        </div>
+    )
+  }
+
+  handleClickLeft(props) {
+    clearInterval(this.int)
+    this.componentDidMount();
```

Comment:
You don't want to directly call the life cycle methods, those are for React to use. You should define your own method and use that instead.